pandas/conftest.py

```diff
@@ -1305,7 +1304,6 @@ def string_storage(request):
 
     * 'python'
     * 'pyarrow'
-    * 'pyarrow_numpy'
```

Comment:
So the `string_storage` fixture will no longer cover the "pyarrow + NaN" variant of the string dtype directly with this change, but:

- Most tests using this fixture are inside the `pandas/tests/io` submodule, which had an override of this fixture to not include "pyarrow_numpy" (and I removed this override)
- I checked/updated the other tests and none of them really need this.